src/OpenTK.Windowing/NativeWindow.cs

```diff
@@ -62,60 +57,241 @@ public bool Visible
         {
             get
             {
-                unsafe
-                {
-                    return glfw.GetWindowAttrib(windowPtr, (int)WindowHint.GLFW_VISIBLE) != 0;
-                }
+                unsafe { return glfw.GetWindowAttrib(windowPtr, (int)WindowHint.GLFW_VISIBLE) != 0; }
             }
             set
             {
                 if (value)
                 {
-                    unsafe
+                    unsafe { glfw.ShowWindow(windowPtr); }
+                }
+                else
+                {
+                    unsafe { glfw.HideWindow(windowPtr); }
+                }
+            }
+        }
+
+        // TODO: Probably not a complete check, this isn't a strong enough guarantee that the window is initialized, has been created and not destroyed (or being destroyed)
+        public bool Exists
+        {
+            get { unsafe { return windowPtr != null; } }
+        }
+
+        public IWindowInfo WindowInfo => throw new NotImplementedException("IWindowInfo is obsolete because platforms are going away, though we could create better borders around GLFW and define 'GLFWWindowInfo : IWindowInfo' but I think it's a case of YAGNI");
+
+        public WindowState WindowState
+        {
+            get
+            {
+                unsafe
+                {
+                    if (glfw.GetWindowAttrib(windowPtr, (int)WindowHint.GLFW_ICONIFIED) == 1)
+                    {
+                        return WindowState.Minimized;
+                    }
+                    else if (glfw.GetWindowMonitor(windowPtr) != null)
                     {
-                        glfw.ShowWindow(windowPtr);
+                        return WindowState.Fullscreen;
+                    }
+                    else
+                    {
+                        var monitor = glfw.GetPrimaryMonitor();
+                        var mode = glfw.GetVideoMode(monitor);
+
+                        int windowWidth, windowHeight;
+                        glfw.GetWindowSize(windowPtr, out windowWidth, out windowHeight);
+
+                        if (mode->width == windowWidth && mode->height == windowHeight)
+                        {
+                            return WindowState.Maximized;
+                        }
                     }
+                    return WindowState.Normal;
                 }
-                else
+            }
+            set
+            {
+                switch (value)
+                {
+                    case WindowState.Normal:
+                        unsafe { glfw.RestoreWindow(windowPtr); }
```

Comment:
here the unsafe code block could be used for the whole property as well, as pretty much any code executed is unsafe here as well..

Reply:
Done.